### DISCUSSION TITLE
lots of "NA" values after preprocessing by "sesame" compared to "minfi/ENmix"

### DISCUSSION THREAD

**Discussion by yazdan59** [zwdzwd/sesame#182] [General]
Hello,
I did a preprocessing step on my methylation data using two different methods (one using "_sesame_" and the other using "_minfi_" and "_ENmix_" packages) to compare the results. 
However, in terms of final **Beta values**, I see a big difference in **"NA"** values. Could anyone explain why "_sesame_" has such many **"NA"** values and what is the best strategy to remove them after preprocessing (because if I remove their corresponding rows, it remains only 365,218 probes out of 866,553)?

Here are my used commands on the input data:

```
# Method 1) "minfi"/"ENmix" Packages
load("RGset_1752.RData"))
# Performing "Illumina+RCP" method
Mset_illumina <- preprocessIllumina(RGset)
betaIllumina_rcp <- rcp(Mset_illumina)
# Create the MethylSet
Mset <- preprocessRaw(RGset)
# Create the RatioSet
Rset <- ratioConvert(Mset)
# Retrieve the Beta-values
bVals= Rset@assays@data@listData[["Beta"]]
dim(bVals)
# [1] 716145   1752
sum(is.na(bVals))
# [1] 0
min(bVals)
# [1] 0
max(bVals)
# [1] 0.999969

# Method 2) "sesame" Packages
load("SigDF_1752.RData"))
sdfs_norm <- openSesame(sdfs, prep="QCDPB", func=NULL)
sdfs_norm_betas <- lapply(sdfs_norm, getBetas)
sdfs_norm_betas_matrix <- do.call(cbind, sdfs_norm_betas)
dim(sdfs_norm_betas_matrix)
# [1] 866553     1752
sum(is.na(sdfs_norm_betas_matrix))
# [1] 278742766
sum(is.na(sdfs_norm_betas_matrix))/length(sdfs_norm_betas_matrix)
# [1] 0.1836007
min(sdfs_norm_betas_matrix)
# [1] NA
max(sdfs_norm_betas_matrix)
# [1] NA
```

Thanks,

**Comment by zwdzwd** [comment]
My guess is that it is related to the low detection success of your data. Sesame has a more stringent detection masking. If you can run the risk of relaxing this threshold, you can tweak the pOOBAH p-value threshold or use the new ELBAR method for detection. Or you can turn off masking fully, by using the `mask=F` in openSesame.

**Comment by yazdan59** [comment]
Thanks for your suggestion. 
I try two scenarios: 1) using the ELBAR method and 2) running the preprocess (**QCDPB**) with _no mask_ option in **openSesame**. And I will let you know about the results.
Just for using the ELBAR method, where is the best place to put its step ("**I**") into the "**QCDPB**" code? What is your recommendation?

**Comment by zwdzwd** [comment]
Sure. 
**I** should replace **P** 
so QCDIB.
I experimented with different orders but didn't notice a big difference in practice.

**Comment by yazdan59** [comment]
Hello again,
As we discussed, I tried two scenarios: 1) using the **ELBAR** method ("**QCDIB**") and 2) running the "**QCDPB**" and then obtaining beta with no mask option. Here are the results and also the result of "**minfi**" package:

```
# "sesame" Packages (first approach)
sdfs_norm_v2 <- openSesame(sdfs, prep="QCDIB", func=NULL)
sdfs_norm_v2_betas <- lapply(sdfs_norm_v2, getBetas)
sdfs_norm_v2_betas_matrix <- do.call(cbind, sdfs_norm_v2_betas)
dim(sdfs_norm_v2_betas_matrix)
# [1] 866553     1752
sum(is.na(sdfs_norm_v2_betas_matrix))
# [1] 185072778
min(sdfs_norm_v2_betas_matrix)
# [1] NA
max(sdfs_norm_v2_betas_matrix)
# [1] NA

# "sesame" Packages (second approach)
sdfs_norm <- openSesame(sdfs, prep="QCDPB", func=NULL)
sdfs_norm_v3_betas <- lapply(sdfs_norm, function(x) getBetas(x, mask=FALSE))
sdfs_norm_v3_betas_matrix <- do.call(cbind, sdfs_norm_v3_betas)
dim(sdfs_norm_v3_betas_matrix)
# [1] 866553     1752
sum(is.na(sdfs_norm_v3_betas_matrix))
# [1] 0
min(sdfs_norm_v3_betas_matrix)
# [1] 0.00370823
max(sdfs_norm_v3_betas_matrix)
# [1] 0.9977153

# "minfi"/"ENmix" Result
Mset_illumina <- preprocessIllumina(RGset)
betaIllumina_rcp <- rcp(Mset_illumina)
Mset <- preprocessRaw(RGset)
Rset <- ratioConvert(Mset)
bVals= Rset@assays@data@listData[["Beta"]]
dim(bVals)
# [1] 866553     1752
sum(is.na(bVals))
# [1] 0
min(bVals)
# [1] 0
max(bVals)
# [1] 0.999969
```
So, as you can see, running the **ELBAR** method caused less "_NA_" values (_185,072,778_) compared to "**QCDPB**" strategy (_278,742,766_). Also, using "**no mask**" option removed all "_NA_" values. So, it seems I should use "**no mask**" option to go to the next level analyses (calculating M-values, DMR, etc.). However, I do not know if in this situation, still we have used the strength of the **QCDPB** sesame preprocessing as mentioned in the papers which recommended its usage compared to other preprocessed methods (such as Illumina, ENmix, noob, rcp, etc.).

**Reply by zwdzwd** [reply]
Hi @yazdan59, Great. The background subtraction and dye bias correction should've been completed even though the problematic probes remain not masked. Just take caution interpreting the methylation signals from ones that are supposed to be masked.

**Reply by yazdan59** [reply]
Thanks for all your suggestions and consideration.
